### ISSUE TITLE
Cards are not generated with the Obsidian plugin

### ISSUE THREAD

**Issue by uuuser-name** [Pseudonium/Obsidian_to_Anki#159]
When I run the vault scanning in the Obsidian plugin, no card get generated.

This is the error message, at the bottom of the console. 

```
Uncaught (in promise) TypeError: Cannot read property 'path' of null
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at a (app.js:1)
    at t.readBinary (app.js:1)
    at FileManager.requests_1 (eval at <anonymous> (app.js:1), <anonymous>:30578:51)
    at async MyPlugin.scanVault (eval at <anonymous> (app.js:1), <anonymous>:30806:9)
    at async HTMLDivElement.eval (eval at <anonymous> (app.js:1), <anonymous>:30842:13)
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
t.readBinary @ app.js:1
requests_1 @ VM103:30578
async function (async)
eval @ VM103:30842
```

And this is the entire log: [obsidian.md-1607994686395.log](https://github.com/Pseudonium/Obsidian_to_Anki/files/5692546/obsidian.md-1607994686395.log)

**Comment by Pseudonium** [comment]
Hmm, try updating the plugin version and AnkiConnect

**Comment by uuuser-name** [comment]
I tried updating Anki, reinstalling AnkiConnect, and reinstalling the Obsidian plugin, the error message slightly changed.
It now shows 
```
Uncaught (in promise) TypeError: Cannot read property 'path' of null
    at FileManager.requests_1 (eval at <anonymous> (app.js:1), <anonymous>:30697:78)
    at MyPlugin.scanVault (eval at <anonymous> (app.js:1), <anonymous>:30941:23)
    at async HTMLDivElement.eval (eval at <anonymous> (app.js:1), <anonymous>:30977:13)
```

**Comment by Pseudonium** [comment]
I see. Looking through the code, this seems like an issue where the plugin is attempting to find a media file, but cannot find it. Could you show me the file(s) it's running on to confirm whether this is the case?

**Comment by uuuser-name** [comment]
I tried deleting a few of the images that seems to be the problem, as well as deleting the anki id tags in the obsidian notes (I hope it won't mess up with the plugin), but it didn't change anything.
There are quite a few hundreds of markdown files the plugin is running on, I don't think I can send them all, is there another way?

**Comment by Pseudonium** [comment]
I'll see if I can add proper error handling for this in a future update. Perhaps try adding the files one at a time to a new vault and scanning that to see if you can identify the particular file that's causing the error? Also, what message did that error come immediately after? The other issue with this error had the message immediately after "Adding media file: ..."

**Comment by uuuser-name** [comment]
The error came immediately after a "Adding media file" message too.
`Adding media file:  Pasted image 20201025134412.png` 

Thank you for your time.

**Comment by Pseudonium** [comment]
Right. Does that media file exist?

**Comment by uuuser-name** [comment]
You were right. The console shows this message for every file that doesn't exist anymore. After deleting the markdown/wikilink mentions to those files, the script runs properly. (!)
This is therefore probably the same issue as #162.

As a side note, great handling of the duplicates by generating an error message stating a note is a duplicate instead of adding it as a duplicate.

**Comment by Pseudonium** [comment]
This will be fixed in next update, have already done the commit that fixes this

**Comment by Pseudonium** [comment]
Fixed in 3.4.1
